### ISSUE TITLE
NavigationRegion don't weld when baked with MeshInstance

### ISSUE THREAD

**Issue by PakkyCX** [godotengine/godot#45007]
**Godot version:**

Godot Engine v4.0.dev.custom_build.a7baf0135

**OS/device including version:**

Windows 10

**Issue description:**

When baking a NavigationRegion with a MeshInstance, the regions no longer want to weld, thus the character can't walk over to the neighboring region.

![image](https://user-images.githubusercontent.com/20498502/103987269-31c0e900-51c7-11eb-8de8-c7fed4b91eeb.png)
(Without mesh included in nav baking)

![image](https://user-images.githubusercontent.com/20498502/103987519-8d8b7200-51c7-11eb-915d-c7abbd7a49dc.png)
(With Cube mesh included in nav baking)

**Steps to reproduce:**
1. Open the reproduction project
2. Run the game (Left click around to move the character, expected to move through both regions without issue.)
3. Close, reparent one of the obstacles (Cube/Sphere) to one of the appropriate NavigationRegions and bake it again
4. Run the game again.

**Minimal reproduction project:**

[nav-test.zip](https://github.com/godotengine/godot/files/5785819/nav-test.zip)

**Comment by Calinou** [comment]
cc @AndreaCatania

**Comment by smix8** [comment]
@ PakkyCX
Tested the MRProject, not even the starting situation works in current Godot 4.x (default edge_margin was lowered further recently).

You aligned the base Meshinstances but I noticed to much distance between the two NavigationMeshes for the default NavigationServer3D.edge_connection_margin. As soon as you set the edge_connection_margin appropriated to your scene layout it works again e.g. in the _ready() function of your level.

```
func _ready():
	var navigation_map : RID = get_world_3d().get_navigation_map()
	NavigationServer3D.map_set_edge_connection_margin(navigation_map, 1.5)
```
@ Calinou
I think this issue can be closed.
